### DISCUSSION TITLE
Recognize buttonpress without href

### DISCUSSION THREAD

**Discussion by ghost** [Hieromon/PageBuilder#38] [Q&A]
Hi,

I am setting up a single page to set some parameters within my ESP32.
I have taken the sample code to set some buttons and get a static method called when they are clicked. Sadly, the only way I found to pass which button has been pressed was like in the example, by using the href. 

Code snippet:
```
<body>
	{{SECAUTOBRIGHT_SET}}
	<p>Lavalamp Control</p>
	<div class="one">
	<p>autobright level</p>
	<p>
		<a class={{SECAUTOBRIGHT_VERYLOW}} href="/?abr=verylow">very low</a> 
		<a class={{SECAUTOBRIGHT_LOW}} href="/?abr=low">low</a>
		<a class={{SECAUTOBRIGHT_MID}} href="/?abr=mid">mid</a> 
		<a class={{SECAUTOBRIGHT_HIGH}} href="/?abr=high">high</a>
	</p>
</body>
```

{{SECAUTOBRIGHT_SET}} calls a method which evaluates abr and sets an enum parameter accordingly. 
The button - SECAUTOBRIGHT_ - methods read which enum was set last and just returns the style for the button (f.e. String "button", so the style is taken from same char array as in the example). This way I can see which button was pressed last.
First question is: is this a good way to accomplish this?

And my problem: the href is then kept within the browser, f.e. when pressed button verylow, the URL reads "http://192.168.178.170/?abr=verylow".
Problem is, when refreshing the page, the button is "pressed" again. Especially when connecting from multiple devices, this is very annoying.
Is there an other way to recognize a buttonpress while staying on the same site, without "keeping" the press within the URL?

**Comment by Hieromon** [comment]
Is that the question of how to apply PageBuilder? Or what are the requirements for HTML code to conform to the characteristics of the ESP32 WebServer class? Is the question What is the best way to do this?
I am not sure where the problem you are having is focused on PageBuilder's functionality.

> I have taken the sample code to set some buttons and get a static method called when they are clicked. Sadly, the only way I found to pass which button has been pressed was like in the example, by using the href.

I am not sure what you are trying to say in the first place. There are other HTML tags that can trigger page transitions besides the `<a>` tag that would be appropriate to fit your requirements. And simple JavaScript can be included in the content if needed.

> Is there an other way to recognize a buttonpress while staying on the same site, without "keeping" the press within the URL?

I am not even sure what the question of this is. Do you mean that the server-side processing is tied across the client's session?
Refs: https://developer.mozilla.org/en-US/docs/Web/HTTP/Session
